### PR TITLE
Header X-Galeb-Error: Externalized to listeners

### PR DESCRIPTION
**Pull request by tuxmonteiro**

### CODE REVIEW

File: router/src/main/java/io/galeb/router/handlers/completionListeners/ProcessorLocalStatusCode.java [hunk 1]
```diff
@@ -50,4 +55,9 @@ default int getFakeStatusCode(final String backend,
 
         return statusLogged;
     }
+
+    String extractUpstreamField(final HeaderMap responseHeaders, String tempRealDest) {
+        final String headerGalebError = responseHeaders != null ? responseHeaders.getFirst(X_GALEB_ERROR) : null;
+        return tempRealDest != null ? tempRealDest : (headerGalebError != null ? headerGalebError : UNKNOWN_TARGET);
```
Comment (igorfernandes) [inline router/src/main/java/io/galeb/router/handlers/completionListeners/ProcessorLocalStatusCode.java]: Why this can't be resume in one line?

Comment (tibraga) [inline router/src/main/java/io/galeb/router/handlers/completionListeners/ProcessorLocalStatusCode.java]: I like explicit code. I would do below:

```
String returnValue = tempRealDest;
final String headerGalebError = responseHeaders != null ? responseHeaders.getFirst(X_GALEB_ERROR) : null;
if (returnValue == null) {
     returnValue = headerGalebError != null ? headerGalebError : UNKNOWN_TARGET
}
return returnValue;
```

Comment (tuxmonteiro) [inline router/src/main/java/io/galeb/router/handlers/completionListeners/ProcessorLocalStatusCode.java]: @igorfernandes , explicit code is more readability. One shot line is not. 
Remembering Martin Fowler (Clean Code book): "Any fool can write code that a computer can understand. Good programmers write code that humans can understand."

Comment (tuxmonteiro) [inline router/src/main/java/io/galeb/router/handlers/completionListeners/ProcessorLocalStatusCode.java]: @tibraga , I agree, but that approach has performance problem:  Even if temRealDest is not null, the responseHeaders will be calling

Comment (tuxmonteiro) [inline router/src/main/java/io/galeb/router/handlers/completionListeners/ProcessorLocalStatusCode.java]: @tibraga , seeing again my code, it has a performance problem:  Even if temRealDest is not null, the responseHeaders will be calling (unnecessary in this case). I will update to better approach.

**Pull request merged**